src/main/java/com/example/nlearn/services/TestService.java

```diff
@@ -57,9 +61,13 @@ public ResponseEntity createTest(FullTest fullTest) {
     }
 
 
-    public ResponseEntity updateTest(Integer id, FullTest fullTest) {
+    public ResponseEntity updateTest(Integer testId, FullTest fullTest, Integer userId, Boolean isAdmin) {
+
+        if (!(isAdmin || testRepository.getById(testId).getAuthor().getId() == userId))
```

Comment:
You should move long conditions to separate methods for readability

Reply:
done